components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -152,9 +152,11 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
                 } else if (errorCode.equals(
                         IdentityCoreConstants.ADMIN_FORCED_USER_PASSWORD_RESET_VIA_OTP_ERROR_CODE)) {
                     String username = request.getParameter(BasicAuthenticatorConstants.USER_NAME);
+                    String tenantDoamin = MultitenantUtils.getTenantDomain(username);
                     String redirectURL = (PASSWORD_RESET_ENDPOINT + queryParams) +
-                            BasicAuthenticatorConstants.USER_NAME + "=" + URLEncoder.encode(username) +
-                            "&confirmation=" + password;
+                            BasicAuthenticatorConstants.USER_NAME_PARAM + URLEncoder.encode(username, BasicAuthenticatorConstants.UTF_8) +
+                            BasicAuthenticatorConstants.TENANT_DOMAIN_PARAM + URLEncoder.encode(tenantDoamin, BasicAuthenticatorConstants.UTF_8) +
+                            BasicAuthenticatorConstants.CONFIRMATION_PARAM + password;
```

Comment:
is password value URL safe?

Reply:
In this case this is not the password. its a used confirmation code.

Reply:
shouldn't we URL encode it as well?